Modeller.fs

```diff
@@ -805,8 +810,29 @@ and modelBoolExpr
                     | Eq -> mkEq
                     | Neq -> mkNeq
                     | _ -> failwith "unreachable[modelBoolExpr::AnyIn]"
+                (* If at least one of the operands is a symbol, we need to
```

Comment:
This new code just tries to make sure that, if one side of an equality/nonequality is a symbol, we try work out its type by modelling the other side first.

This is unrelated to atomic conditionals, I think it's wormed its way into here due to some hacking I was doing around CAV.